### ISSUE TITLE
Grails 3.0.1: NotifyOfEvent and DomainClass plugin

### ISSUE THREAD

**Issue by Sabst** [grails/grails-core#639]
This used to work with 2.3 and 2.4 versions.
This fails with version 3, incl. 3.0.1.

Regards.

```
// This is a simple console script
// MongoDB is used as the datasource
GroovyClassLoader gcl = new GroovyClassLoader(this.getClass().getClassLoader())
def clazzDomain = gcl.parseClass('''
@grails.persistence.Entity
class SomeDomain {
    String name
}
''')
def pluginManager = ctx.getBean('pluginManager')
def dc = pluginManager.getGrailsPlugin("domainClass")
dc.notifyOfEvent(grails.plugins.GrailsPlugin.EVENT_ON_CHANGE, clazzDomain)

/*
java.lang.NullPointerException
	at grails.plugins.Plugin.beans(Plugin.groovy:159)
	at grails.plugins.Plugin$beans.callCurrent(Unknown Source)
	at org.grails.plugins.domain.DomainClassGrailsPlugin.onChange(DomainClassGrailsPlugin.groovy:117)
	at org.grails.plugins.DefaultGrailsPlugin.notifyOfEvent(DefaultGrailsPlugin.java:787)
	at grails.plugins.GrailsPlugin$notifyOfEvent.call(Unknown Source)
	at ConsoleScript1.run(ConsoleScript1:11)
	at org.springsource.loaded.ri.ReflectiveInterceptor.jlrMethodInvoke(ReflectiveInterceptor.java:1270)
*/
```

**Comment by jeffbrown** [comment]
We will look into why that is happening.  Can you summarize why it is that you are doing this?  Thanks for the feedback.

**Comment by jeffbrown** [comment]
This may be related to the fact that at https://github.com/grails/grails-core/blob/9f78cdf17e140de37cfb5de6671131df3606f2fe/grails-plugin-domain-class/src/main/groovy/org/grails/plugins/domain/DomainClassGrailsPlugin.groovy#L47 `DomainClassGrailsPlugin` defines a `GrailsApplication` property which overrides the corresponding property defined in the parent class at https://github.com/grails/grails-core/blob/9f78cdf17e140de37cfb5de6671131df3606f2fe/grails-core/src/main/groovy/grails/plugins/Plugin.groovy#L47.  Because of that, under certain circumstances the one in the parent class isn't being initialized.

**Comment by Sabst** [comment]
I think there is a lot of potential in meta-programming at the framework level.
After (painfully) following the work made by Burt, Lim Chee Kin and many others in this area, I was quite happy with the refactoring the team made in recent versions (all those internal plugins you created are such an impressive work).
Do you see any workaround for this problem or do you think I will have to wait for a fix?

**Comment by jeffbrown** [comment]
> I think there is a lot of potential in meta-programming at the framework level.

I agree but it isn't clear what that has to do with the issue at hand here.   Can you summarize why it is that you are doing this? Thanks for the feedback.

**Comment by graemerocher** [comment]
closing, no feedback provided
